tests/test_distributed.py

```diff
@@ -53,185 +54,149 @@ def calculate_entropy_full(logits):
-            entropy_full = calculate_entropy_full(full_logits)
+    fake_parallel_logits = full_logits.chunk(world_size, dim=-1)[rank].detach().clone().requires_grad_()
 
-            assert torch.allclose(
-                entropy_distributed, entropy_full
-            ), "entropy between distributed and full path are different!"
+    with torch.no_grad():
+        entropy_distributed = calculate_distributed_entropy(fake_parallel_logits)
+        entropy_full = calculate_entropy_full(full_logits)
 
-        calculate_entropy_full(full_logits).sum().backward()
-        grad_full_slice = full_logits.grad.chunk(world_size, dim=-1)[rank]
+        assert torch.allclose(
+            entropy_distributed, entropy_full
+        ), "entropy between distributed and full path are different!"
 
-        full_logits.grad = None
+    calculate_entropy_full(full_logits).sum().backward()
+    grad_full_slice = full_logits.grad.chunk(world_size, dim=-1)[rank]
 
-        calculate_distributed_entropy(fake_parallel_logits).sum().backward()
-        grad_distributed = fake_parallel_logits.grad
+    full_logits.grad = None
 
-        assert torch.allclose(
-            grad_full_slice, grad_distributed
-        ), "grad of entropy between distributed and full path are different!"
-
-    @pytest.mark.run_only_on("GPU")
-    def test_distributed_masked_global_mean_var(self):
-        self._run_test(self._test_masked_global_mean_var)
-
-    @pytest.mark.run_only_on("GPU")
-    @pytest.mark.parametrize(
-        "batch_size,seed,dtype,atol,rtol,higher_stability",
-        [
-            (1, 9999, torch.float32, 1e-08, 1e-05, False),
-            (4, 100, torch.float32, 1e-08, 1e-05, False),
-            (8, 1234, torch.float32, 1e-08, 1e-05, False),
-            (1, 9999, torch.float32, 1e-08, 1e-05, True),
-            (4, 100, torch.float32, 1e-08, 1e-05, True),
-            (8, 1234, torch.float32, 1e-08, 1e-05, True),
-            (1, 746, torch.bfloat16, 0.005, 0.01, False),
-            (4, 334, torch.bfloat16, 0.005, 0.01, False),
-            (8, 123456, torch.bfloat16, 0.005, 0.01, False),
-            (1, 746, torch.bfloat16, 0.005, 0.01, True),
-            (4, 334, torch.bfloat16, 0.005, 0.01, True),
-            (8, 123456, torch.bfloat16, 0.005, 0.01, True),
-        ],
-    )
-    def test_distributed_log_probs(self, batch_size, seed, dtype, atol, rtol, higher_stability):
-        self._run_test(self._test_distributed_log_probs, batch_size, seed, dtype, atol, rtol, higher_stability)
+    calculate_distributed_entropy(fake_parallel_logits).sum().backward()
+    grad_distributed = fake_parallel_logits.grad
 
-    @pytest.mark.run_only_on("GPU")
-    @pytest.mark.parametrize("batch_size,seed", [(1, 5555), (4, 6666)])
-    def test_distributed_entropy(self, batch_size, seed):
-        self._run_test(self._test_distributed_entropy, batch_size, seed)
+    assert torch.allclose(
```

Comment:
can we use torch's API? https://pytorch.org/docs/stable/testing.html#torch.testing.assert_close